ldms/src/test/Makefile.am

```diff
@@ -39,4 +39,29 @@ test_metric_SOURCES = test_metric.c
 test_metric_LDADD = -lldms
 test_metric_LDFLAGS = $(AM_LDFLAGS) -pthread -lm
 
-TESTS = $(check_PROGRAMS)
+EXTRA_DIST=ldms-run-static-tests.sh.in
+EXTRA_DIST+=ldms-static-test-list.sh.in
+EXTRA_DIST+=ldms-static-test-bypass.in
+
+CLEANFILES=ldms-run-static-tests.sh
+CLEANFILES+=ldms-static-test-list.sh
+CLEANFILES+=ldms-static-test-bypass 
+
+%.sh: %.sh.in
+	$(do_subst) < $< > $@
+	chmod 755 $@
+
+installcheck-local: ldms-run-static-tests.sh
+
+install-data-hook: $(CLEANFILES)
+	$(MKDIR_P) $(DESTDIR)$(pkglibdir)
+	$(INSTALL) -m 755 ldms-static-test-list.sh $(DESTDIR)$(pkglibdir)
+	$(INSTALL) -m 755 ldms-run-static-tests.sh $(DESTDIR)$(pkglibdir)
+	$(INSTALL) -m 644 ldms-static-test-bypass $(DESTDIR)$(pkglibdir)
+
+uninstall-hook:
+	rm $(DESTDIR)$(pkglibdir)/ldms-static-test-list.sh
+	rm $(DESTDIR)$(pkglibdir)/ldms-run-static-tests.sh
+	rm $(DESTDIR)$(pkglibdir)/ldms-static-test-bypass
+
+TESTS = $(check_PROGRAMS) ldms-run-static-tests.sh
```

Comment:
I am not sure what you are going for here. Is the ldms-run-static-test.sh script supposed to run at "make check" time, or at "make installcheck" time? Including it in "TESTS" makes it run at "make check" time.

Reply:
The ldms-run-static-test.sh is supposed to run at "make installcheck" time. Make installcheck tests the installed version of ldms-run-static-test.sh so would I need to provide the location of this installed file (i.e. $pkglibdir/ldms-run-static-test)?

Reply:
Also, I'm trying to install the scripts into the $pkglibdir which is why there is the install-data-hook: section (56-60).

Reply:
"TESTS" does not run scripts at installcheck time, to the best of my knowledge. It _does_ run things at "make check" time. Your script has hard coded paths to install locations, to I'm guessing it won't work at check time, only at installcheck time. So you'll need to remove it from TESTS and explicitly run it under the installcheck-local target.

We don't need to us install-data-hook just to install a file, but I know that is an oddity in other places in the ldms tree as well. Some day I might get around to fixing that.

Reply:
If you were asking about where you need the $pkglibdir path prefix under installcheck-local, then no, I don't think so. You can run the ldms-run-static-test.sh from the source tree, and have it test against the installed version of LDMS. This assumes that you wrote ldms-run-static-test.sh to not care about where is is run from. I haven't really reviewed the internals of that script.

The only reason why you would install the ldms-run-static-test.sh is if someone plans to run that script in situations where the source tree is no longer available (and therefore no possibility of running "make installcheck").

Reply:
Just saw this. No, the ldms-run-static-test doesn't care where it's at but it does look for the location in which ldms-static-test-list.sh and ldms-static-test-bypass are located (which right now is set to $pkglibdir).